editor/code_editor.cpp

```diff
@@ -1655,6 +1655,12 @@ void CodeTextEditor::_notification(int p_what) {
 			warning_button->set_icon(get_theme_icon("NodeWarning", "EditorIcons"));
 			add_theme_constant_override("separation", 4 * EDSCALE);
 		} break;
+		case NOTIFICATION_EXIT_TREE: {
+			if (find_replace_bar) {
+				remove_child(find_replace_bar);
+				find_replace_bar = nullptr;
+			}
+		} break;
```

Comment:
Couldn't it be deleted here once instead of having to do it both in ScriptEditorPlugin and ShaderEditorPlugin?

Reply:
The problem is that code editor might be freed when closing script (not 100% sure when this happens though, as the crash is rather random). This is what was causing the bar being freed and the crash. I'm removing the bar to avoid freeing it. Script editor still has a reference, for shader editor I had to do a little workaround.

Reply:
I wonder if we're not getting more troubles than gain by reusing the same `FindReplaceBar` in different editors. It would be cleaner if there was one instance in each plugin as child of the plugin, which gets freed with it.